### PR TITLE
CVE-2020-16137

### PR DESCRIPTION
**Pull request by debifrank**
The introduces a new module coinciding with CVE-2020-16137. This vulnerability allows an attacker to change the SSH administrative credentials to any values.

## Verification

List the steps needed to make sure this thing works

- [ ] Start `msfconsole`
- [ ] `use exploit/linux/ssh/CVE-2020-16137`
- [ ] Supply RHOST
- [ ] Supply USER
- [ ] Supply PASS
- [ ] `run` 
- [ ] **Verify** `Attempting to set SSH credentials.`
- [ ] **Verify** If device does not have Web Access enabled: `Device doesn't appear to be functioning or web access is not enabled.`
- [ ] **Verify** If device does have Web Access enabled: `SSH attack finished!`
- [ ] **Verify** If device does have Web Access enabled: `Try to login using the supplied credentials $USER:$PASS`
- [ ] **Verify** If device does have Web Access enabled: `You must specify the key exchange when connecting or the device will be DoS'd!`
- [ ] **Verify** If device does have Web Access enabled: `ssh -oKexAlgorithms=+diffie-hellman-group1-sha1 $USER@$PASS`
- [ ] **Document** Vulnerability documentation can be found [here](https://blacklanternsecurity.com/2020-08-07-Cisco-Unified-IP-Conference-Station-7937G/)

Demos on our blog contain manual exploitation. If video demonstrations of the modules are necessary they can be provided.

### CODE REVIEW

**Comment by gwillcox-r7** [comment]
Took a quick look over the code and it looks like this is just making a HTTP request in Python. It would be advisable given these circumstances to just port this code over to Ruby instead. You can find a guide on how to do this at https://github.com/rapid7/metasploit-framework/wiki/How-to-Send-an-HTTP-Request-Using-HTTPClient

**Comment by gwillcox-r7** [comment]
Please also not that for this and all your other PRs that using terminology like `FIRING ZE MIZZLES!` whilst funny and in good jest, doesn't help our users understand what is going on. Please replace these outputs with more accurate statements that describe what is going on at that particular point in the code.

**Comment by debifrank** [comment]
@gwillcox-r7 I was hoping to provide these as examples for the recent Python support in place. If it is absolutely necessary I can convert them to Ruby. I'm happy to change my jest. :)

**Comment by debifrank** [comment]
It seems the sanity check failed after only changing the log info when it passed initially. Looking through the supplied log file, I'm unsure what went wrong.

**Comment by bcoles** [comment]
> 
> 
> It seems the sanity check failed after only changing the log info when it passed initially. Looking through the supplied log file, I'm unsure what went wrong.

The sanity check is insane. It can often be safely ignored.

**Review by space-r7 — COMMENTED** [review]
Hi @debifrank, thank you for the module!

We don't *typically* use the CVE for the name of the module / documentation, although it might be included as part of the name in some cases. Please rename your module and documentation to be more descriptive of what the module exploits, like `cisco_7937g_ssh_privesc.rb` or something similar.

**Comment by debifrank** [comment]
Changes have been put through on all three module PRs. Let me know if there is anything else needed.

File: documentation/modules/exploit/linux/ssh/cisco_7937g_ssh_privesc.md [hunk 1]
```diff
@@ -0,0 +1,448 @@
...
+
+## Scenarios
+
+### Cisco 7937G Running Firmware Version SCCP-1-4-5-7
+
+#### Successful Scenario
+
+```
+msf5 > use exploit/linux/ssh/cve_2020_16137 
+msf5 auxiliary(linux/ssh/cve_2020_16137) > set user test
```
Comment (space-r7) [inline documentation/modules/exploit/linux/ssh/cisco_7937g_ssh_privesc.md:40]: Is this output from another module or from this one?

**Review by space-r7 — COMMENTED** [review]
It looks like the documentation might need to be updated given the name change and some mismatches with the output. Also, does this module get a session on the device or does it create a user account? If it's the latter, I think this should probably be listed as an auxiliary module instead.

File: documentation/modules/exploit/linux/ssh/cisco_7937g_ssh_privesc.md (hunk 1 above)
Comment (debifrank) [reply documentation/modules/exploit/linux/ssh/cisco_7937g_ssh_privesc.md:40]: From this one. Something about my copy of modules got borked. I'll redo this section.

File: documentation/modules/exploit/linux/ssh/cisco_7937g_ssh_privesc.md [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+### Description
+
+This module exploits a feature that should not be available via the web interface. An unauthenticated user may set the credentials for SSH access to any username and password combination desired, giving access to administrative functions through an SSH connection.
+
+## Verification Steps
+
+  1. Obtain a Cisco 7937G Conference Station.
+  2. Enable Web Access and SSH Access on the device.
+  3. Start msfconsole
+  4. Do: `use auxiliary/dos/cisco/CVE-2020-16139`
```
[the remark this answers is not in the document]
Comment (debifrank) [reply documentation/modules/exploit/linux/ssh/cisco_7937g_ssh_privesc.md:15]: I'll edit the documentation to reflect the name changes.

**Comment by debifrank** [comment]
This module sets the SSH credentials for the device. The module does not spawn the connection afterward. If it is more reasonable to place it in auxiliary, which directory exactly should it be placed?

**Comment by space-r7** [comment]
> This module sets the SSH credentials for the device. The module does not spawn the connection afterward. If it is more reasonable to place it in auxiliary, which directory exactly should it be placed?

Since the exploit works over http, it'd probably best fit in the `modules/auxiliary/admin/http/` path.

File: documentation/modules/exploit/linux/ssh/cisco_7937g_ssh_privesc.md (hunk 1 above)
Comment (space-r7) [reply documentation/modules/exploit/linux/ssh/cisco_7937g_ssh_privesc.md:40]: Thanks much!

**Pull request merged**